crates/db_schema/src/traits.rs

```diff
@@ -213,3 +213,16 @@ pub trait InternalToCombinedView {
   /// Maps the combined DB row to an enum
   fn map_to_enum(self) -> Option<Self::CombinedView>;
 }
+
+pub trait PaginationCursorBuilder {
+  /// Builds a pagination cursor for the given query result.
+  fn cursor(&self) -> PaginationCursor;
+}
+
+#[async_trait]
+pub trait PaginationCursorReader {
+  /// Reads a database row from a given pagination cursor.
+  async fn from_cursor(cursor: &PaginationCursor, conn: &mut DbPool<'_>) -> LemmyResult<Self>
+  where
+    Self: Sized;
+}
```

Comment:
You can remove async_trait macro here and use Rusts builtin support for async methods in traits.

Reply:
Done.